cli/rt/11_url.js

```diff
@@ -876,6 +900,61 @@
     );
   }
 
+  function encodeStr(str, noEscapeTable, hexTable) {
```

Comment:
Can you call this `encodeUrlSearchParam()`? Also I don't think this should take these second and third arguments in favour of referring to them as globals.